### PR TITLE
metrics: update monitoring template in grafana

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve? 
- Add TSO handle duration in server-side
- Add Heartbeat QPS for `update_kv` and `update_cache`
- Add direction of balance hot region
- Add the hot cache entry numbers 
- Fix balance Region movement
- Add Rule checker 

### What is changed and how it works?
Update `promQL`

### Check List 

Tests 

 - Mannual test
image

### Release note 
- No

### CODE REVIEW

File: metrics/grafana/pd.json [hunk 1]
```diff
@@ -4328,7 +4813,7 @@
           "timeFrom": null,
           "timeRegions": [],
           "timeShift": null,
-          "title": "Store write rate keys",
+          "title": "HotCache Read entry numbers",
```
Comment (rleungx) [inline metrics/grafana/pd.json:4816]:
```suggestion
          "title": "Hot cache read entry numbers",
```

File: metrics/grafana/pd.json [hunk 2]
```diff
@@ -2774,1024 +3379,1001 @@
...
+          "legendFormat": "{{store}}",
+          "refId": "A",
+          "step": 4
+        }
+      ],
+      "thresholds": [],
+      "timeFrom": null,
+      "timeRegions": [],
+      "timeShift": null,
+      "title": "HotCache Write entry numbers",
```
Comment (rleungx) [inline metrics/grafana/pd.json:4031]:
```suggestion
      "title": "Hot cache write entry numbers",
```

File: metrics/grafana/pd.json [hunk 3]
```diff
@@ -6394,19 +7175,29 @@
...
+              "legendFormat": "99.999%  tso",
+              "refId": "B"
             }
           ],
           "thresholds": [],
           "timeFrom": null,
           "timeRegions": [],
           "timeShift": null,
-          "title": "Raft committed index",
+          "title": "PD Server TSO  handle time + Client Revc time",
```
Comment (rleungx) [inline metrics/grafana/pd.json:7198]:
```suggestion
          "title": "PD server TSO handle time + Client recv time",
```
BTW, we need to change the description of this panel

File: metrics/grafana/pd.json [hunk 4]
```diff
@@ -918,923 +918,1327 @@
       }
     },
     {
-      "collapsed": true,
+      "collapsed": false,
```
Comment (rleungx) [inline metrics/grafana/pd.json:921]: It's better to collapse all.

File: metrics/grafana/pd.json [hunk 5]
```diff
@@ -2774,1024 +3379,1001 @@
...
+          "intervalFactor": 1,
+          "legendFormat": "{{type}}-{{store}}",
+          "refId": "B"
+        }
+      ],
+      "thresholds": [],
+      "timeFrom": null,
+      "timeRegions": [],
+      "timeShift": null,
+      "title": "Direction of hotspot move leader",
```
Comment (rleungx) [inline metrics/grafana/pd.json:4228]: The direction is only in hot write row.

Comment (nolouch) [reply metrics/grafana/pd.json:4228]: yes. I think it's about the total hot region rather than hot read or hot write.

**Pull request merged**